File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/Card.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

@Builder
public record Card(
    String id,
    Component media,
    Component headerPrefix,
    Component header,
    Component title,
    Component subtitle,
    Component headerSuffix,
    Component content,
    Component footer,
    List<CardVariant> variants,
    String style,
    String cssClasses)
    implements Component {}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/Card.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import Component from "@mateu/shared/apiClients/dtos/Component";

export default interface Card extends ComponentMetadata {

    title: Component
    subtitle: Component
    media: Component
    headerPrefix: Component
    header: Component
    headerSuffix: Component
    content: Component
    footer: Component
    variants: string[]

}